src/pages/Register.js

```diff
@@ -1,17 +1,332 @@
 import { Helmet } from "react-helmet";
+import { useState, useRef } from "react";
+import { Listbox } from "@headlessui/react";
+import { CheckIcon, SelectorIcon } from "@heroicons/react/solid";
 import Page from "../components/Page";
 
+/**
+ * @desc Registration page where hackers can sign up for the hackathon. After
+ * submitting, the backend is updated and they will recieve a success message
+ * on the page.
+ *
+ * @author Rob
+ */
 const Register = () => {
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [schoolName, setSchoolName] = useState("");
+  const [email, setEmail] = useState("");
+  const [phoneNumber, setPhoneNumber] = useState("");
+  const [resume, setResume] = useState(null);
+  const [graduation, setGraduation] = useState("");
+  const [dietaryRestrictions, setDietaryRestrictions] = useState("");
+  const [github, setGithub] = useState("");
+  const [linkedIn, setLinkedIn] = useState("");
+
+  const trackOptions = ["Beginner", "Advanced"];
+  const [selectedTrack, setSelectedTrack] = useState(trackOptions[0]);
+
+  const attendingOptions = ["In Person", "Virtual"];
+  const [attendingOption, setAttendingOption] = useState(attendingOptions[0]);
+
+  // "unset" | "success" | "failure" | "pending"
+  const [registrationState, setRegistrationState] = useState("unset");
+  const [feedbackMessage, setFeedbackMessage] = useState("");
```

Comment:
Just a recommendation here, but using a form library like [formik](https://formik.org/) may help you reduce the state you need to keep track of in this component.

Reply:
We did look into this; for now we decided to just implement it with a bunch of state and take a "wait and see" approach to see if it is a problem.